src/main/java/transforms/SubstringTransform.java

```diff
@@ -4,30 +4,45 @@
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.openlattice.shuttle.transformations.Transformation;
 import com.openlattice.shuttle.util.Constants;
+import java.util.Objects;
+
 
 public class SubstringTransform extends Transformation<String> {
     private final int index;
+    private final int endIndex;
 
     /**
      * Represents a transformation to get a substring starting from a certain index (eg 3th character: index = 2).
      *
      * @param index: where to start subsetting if prefix is found
+     * @param endIndex: where to start subsetting if prefix is found
      */
     @JsonCreator
     public SubstringTransform(
-            @JsonProperty( Constants.INDEX ) int index ) {
+            @JsonProperty( Constants.INDEX ) int index,
+            @JsonProperty( Constants.END_INDEX ) int endIndex ) {
         this.index = index;
+        this.endIndex = endIndex;
     }
 
-    @JsonProperty( Constants.LOC )
-    public int getSubstringLocation() {
+    @JsonProperty( Constants.INDEX )
+    public int getIndex() {
         return index;
     }
 
+    @JsonProperty( Constants.END_INDEX )
+    public int getEndIndex() {
+        return endIndex;
+    }
+
     @Override
     public Object applyValue( String o ) {
         final String output;
-        output = o.substring( index );
+        if ( Objects.equals( endIndex, null ) ){
```

Comment:
This will always be false since `int`s can never be null

Reply:
true -- if you want to make it nullable you can make it an `Integer` or an `Optional<Integer>`